src/Dapr.Actors.Generators/ActorClientGenerator.cs

```diff
@@ -20,284 +26,262 @@ namespace Dapr.Actors.Generators;
-            internal sealed class GenerateActorClientAttribute : Attribute
-            {{
-                public string? Name {{ get; set; }}
-
-                public string? Namespace {{ get; set; }}
-            }}
-        }}";
-
-    private sealed class ActorInterfaceSyntaxReceiver : ISyntaxContextReceiver
+    /// <inheritdoc />
+    public void Initialize(IncrementalGeneratorInitializationContext context)
     {
-        private readonly List<INamedTypeSymbol> models = new();
-
-        public IEnumerable<INamedTypeSymbol> Models => this.models;
-
-        #region ISyntaxContextReceiver Members
-
-        public void OnVisitSyntaxNode(GeneratorSyntaxContext context)
+        // Register the source output that generates the attribute definitions for ActorMethodAttribute and GenerateActorClientAttribute.
+        context.RegisterPostInitializationOutput(context =>
         {
-            if (context.Node is not InterfaceDeclarationSyntax interfaceDeclarationSyntax
-                || interfaceDeclarationSyntax.AttributeLists.Count == 0)
-            {
-                return;
-            }
-
-            var interfaceSymbol = context.SemanticModel.GetDeclaredSymbol(interfaceDeclarationSyntax) as INamedTypeSymbol;
-
-            if (interfaceSymbol is null
-                || !interfaceSymbol.GetAttributes().Any(a => a.AttributeClass?.ToString() == GenerateActorClientAttributeFullTypeName))
-            {
-                return;
-            }
-
-            this.models.Add(interfaceSymbol);
-        }
-
-        #endregion
+            context.AddSource(
+                $"{Constants.ActorMethodAttributeFullTypeName}.g.cs",
+                Templates.ActorMethodAttributeSourceText(Constants.GeneratorsNamespace));
+
+            context.AddSource(
+                $"{Constants.GenerateActorClientAttributeFullTypeName}.g.cs",
+                Templates.GenerateActorClientAttributeSourceText(Constants.GeneratorsNamespace));
+        });
+
+        // Register the value provider that triggers the generation of actor clients when detecting the GenerateActorClientAttribute.
+        IncrementalValuesProvider<ActorClientDescriptor> actorClientsToGenerate = context.SyntaxProvider
+            .ForAttributeWithMetadataName(
+                Constants.GenerateActorClientAttributeFullTypeName,
+                predicate: static (_, _) => true,
+                transform: static (gasc, cancellationToken) => CreateActorClientDescriptor(gasc, cancellationToken));
+
+        // Register the source output that generates the actor clients.
+        context.RegisterSourceOutput(actorClientsToGenerate, GenerateActorClientCode);
     }
 
-    #region ISourceGenerator Members
-
-    /// <inheritdoc />
-    public void Execute(GeneratorExecutionContext context)
+    /// <summary>
+    /// Returns the descriptor for the actor client to generate.
+    /// </summary>
+    /// <param name="context"></param>

```

Comment:
Meaningful documentation around parameters, expected returns and exceptions thrown goes a long way to making the library easier to use. Please could you fill in the docs here?

Reply:
Tomorrow i will improve summaries.